lib/MetagenomeUtils/Utils/MetagenomeFileUtils.py

```diff
@@ -643,28 +660,42 @@ def extract_binned_contigs_as_assembly(self, params):
         self._validate_extract_binned_contigs_as_assembly_params(params)
 
         extracted_assemblies = params.get('extracted_assemblies')
-        bin_id_list = []
-        for extracted_assembly in extracted_assemblies:
-            bin_id = extracted_assembly.get('bin_id')[0].strip()
-            bin_id_list.append(bin_id)
+
+        # check for empty list here, if so create one based on all bins from
+        # binned_contig_obj_ref
+        if ( len( extracted_assemblies ) < 1 ):
```

Comment:
I would do `if not extracted_assemblies`

Reply:
I don't see the necessary of this if-else. If extracted_assemblies is empty, all you do is having bin_id_list = None. But bin_id_list is [] by default. I don't see the following code will execute differently with bin_id_list = None or [].